src/ocl_util.cc

```diff
@@ -142,6 +142,26 @@ namespace boda
     out.resize( sz );
     err = clGetDeviceIDs( platform, device_type, sz, &out[0], 0 );
     cl_err_chk( err, "clGetDeviceIDs(out)" );
+  }
+  
+  void cl_get_platform_device_info(cl_platform_id plt, cl_device_id dev) {
+    cl_char *platform_name, *device_name;
+    size_t plt_sz, dev_sz;
+    cl_int err;
+
+    err = clGetPlatformInfo(plt, CL_PLATFORM_NAME, 0, NULL, &plt_sz);
+    cl_err_chk( err, "clGetPlatformInfo(size)" );
+    platform_name = (cl_char*) malloc(plt_sz);
```

Comment:
do not use raw malloc/free, use either smart pointers, strings, vectors, or other mechanisms. also, when calling the opencl API, follow existing patterns for accessing fields and getting values. note that in this case, it appears the malloc()'d strings are simply leaked.

see get_info<>() and get_info_str<>(); if these are currently-unused info sturcts, add wrappers following the pattern for the others. in this case, there is already a Program_t wrapper class for clGetDeviceInfo(); you'll needs to add a new similar one (Platform_t) for clGetPlatformInfo().

in general, when adding small amounts of code to a larger/existing codebase, try to follow existing patterns/conventions.